modules/javafx.graphics/src/main/native-glass/mac/GlassView3D.m

```diff
@@ -810,8 +810,10 @@ - (NSRect) firstRectForCharacterRange:(NSRange)theRange actualRange:(NSRangePoin
     IMLOG("firstRectForCharacterRange called %lu %lu",
           (unsigned long)theRange.location, (unsigned long)theRange.length);
     NSRect result = [self->_delegate getInputMethodCandidatePosRequest:0];
-    NSRect screenFrame = [[NSScreen mainScreen] frame];
-    result.origin.y = screenFrame.size.height - result.origin.y;
+    if (NSScreen.screens.count) {
+        NSRect screenFrame = NSScreen.screens[0].frame;
```

Comment:
I assume this code block is not entered in the case of a single monitor?

Reply:
Yes, it is entered if there's a single monitor (NSScreen.screens.count will be 1). We always have to flip the y coordinate.